### ISSUE TITLE
Why this equation `` hasn't solution. But wolfram has solution

### ISSUE THREAD

**Issue by satels** [sympy/sympy#27322]
```python
>>> from sympy import Eq, sign, Symbol, solveset
>>> x = Symbol('x', real=True)
>>> solveset(Eq(sign(1-1/x**2)/x**3, 0))
Complement(ConditionSet(x, Eq(sign((x**2 - 1)/x**2), 0), Complexes), {0})
```

Wolfram: https://www.wolframalpha.com/input?i=2*sign%283+-+1%2Fx%5E2%29%2Fx%5E3+%3D+0

![Image](https://github.com/user-attachments/assets/3d4fea1e-da62-4841-8ed3-aba5cdb18e72)

**Comment by asmeurer** [comment]
It seems solveset can handle it if it is rewritten in Piecewise form:

```py
>>> solveset(Eq(Piecewise((-1, (3-1/x**2) < 0), (0, Eq(3-1/x**2, 0)), (1, (3-1/x**2) > 0))/x**3, 0), domain=S.Reals)
{-sqrt(3)/3, sqrt(3)/3}
>>> solveset(Eq(Piecewise((-1, (1-1/x**2) < 0), (0, Eq(1-1/x**2, 0)), (1, (1-1/x**2) > 0))/x**3, 0), domain=S.Reals)
{-1, 1}
```

but annoyingly, even `rewrite(Piecewise)` doesn't work:

```py
>>> sign(1-1/x**2).rewrite(Piecewise)
sign(1 - 1/x**2)
```

because it can't determine if the argument is real:

```py
>>> (1-1/x**2).is_extended_real
>>> x.is_real
True
```

**Comment by oscarbenjamin** [comment]
We need something else rather than `is_real`. The semantics of `is_real` are not right for this sort of distinction because it tries conservatively to judge that an expression represents a well defined real number but we don't care about well defined-ness in this case.

**Comment by asmeurer** [comment]
rewrite(Piecewise) should construct Piecewise expressions that are correct for all inputs, but still lets solveset give useful answers. The whole point of Piecewise is to represent conditional expressions symbolically which the old assumptions does not allow.

If SymPy used the convention that `x > y` is False if x or y is nonreal, then `Piecewise((1, x > 0), (-1, x < 0), (0, Eq(x, 0)), (x/abs(x), True))` would be correct and would still let solveset work in this instance. Without that convention, it would have to use `Q.real(x)` or `Contains(x, S.Reals)` in the piecewise conditions, which would probably also make it harder for things like solveset to deal with.

**Comment by asmeurer** [comment]
Actually, this works:

```py
>>> solveset(Eq(sign(1-1/x**2).rewrite(Abs)/x**3, 0), domain=S.Reals)
{-1, 1}
```

Although we also get

```py
>>> solveset(Eq((1-1/x**2)/abs(1 - 1/x**2)/x**3, 0), domain=S.Reals)
EmptySet
```

which is probably coming from solveset not being as careful as solve about how it filters "divide by zero" solutions.

**Comment by oscarbenjamin** [comment]
> rewrite(Piecewise) should construct Piecewise expressions that are correct for all inputs

Yes, but in this case the difficulty in the beginning is in interpretting the meaning of the `sign` function. It can only be written as Piecewise if it is the real sign function. Checking whether the argument is a well defined definitely real number for all possible values of the symbols is not really what we want to check in order to know that it is the real sign function. Instead we want to know that the expression is real for all cases where it is well defined.

**Comment by asmeurer** [comment]
This would also be a simple fix, since `sign.rewrite(Abs)` already returns a Piecewise that is correct for all complex arguments.

```
diff --git a/sympy/functions/elementary/complexes.py b/sympy/functions/elementary/complexes.py
index 27043838d7..e5c90d5aa5 100644
--- a/sympy/functions/elementary/complexes.py
+++ b/sympy/functions/elementary/complexes.py
@@ -430,6 +430,7 @@ def _eval_nseries(self, x, n, logx, cdir=0):
     def _eval_rewrite_as_Piecewise(self, arg, **kwargs):
         if arg.is_extended_real:
             return Piecewise((1, arg > 0), (-1, arg < 0), (0, True))
+        return self._eval_rewrite_as_Abs(arg, **kwargs)

     def _eval_rewrite_as_Heaviside(self, arg, **kwargs):
         from sympy.functions.special.delta_functions import Heaviside
```
